activerecord/test/cases/fixtures_test.rb

```diff
@@ -79,6 +79,56 @@ def test_bulk_insert
         ActiveSupport::Notifications.unsubscribe(subscription)
       end
     end
+
+    def test_bulk_insert_multiple_table_with_a_multi_statement_query
+      subscriber = InsertQuerySubscriber.new
+      subscription = ActiveSupport::Notifications.subscribe("sql.active_record", subscriber)
+
+      create_fixtures("bulbs", "authors", "computers")
+
+      expected_sql = <<-EOS.strip_heredoc
+        INSERT INTO #{ActiveRecord::Base.connection.quote_table_name("bulbs")} .*
+        INSERT INTO #{ActiveRecord::Base.connection.quote_table_name("authors")} .*
+        INSERT INTO #{ActiveRecord::Base.connection.quote_table_name("computers")} .*
+      EOS
+      assert_equal 1, subscriber.events.size
+      assert_match(/#{expected_sql}/, subscriber.events.first)
+    ensure
+      ActiveSupport::Notifications.unsubscribe(subscription)
+    end
+
+    def test_bulk_insert_with_a_multi_statement_query_raises_an_exception_when_any_insert_fails
+      require "models/aircraft"
+
+      assert_equal false, Aircraft.columns_hash["wheels_count"].null
+      fixtures = {
+        "aircraft" => [
+          { "name" => "working_aircrafts", "wheels_count" => 2 },
+          { "name" => "broken_aircrafts", "wheels_count" => nil },
+        ]
+      }
+
+      assert_no_difference "Aircraft.count" do
+        assert_raises(ActiveRecord::NotNullViolation) do
+          ActiveRecord::Base.connection.insert_fixtures_set(fixtures)
+        end
+      end
+    end
+  end
+
+  def test_insert_fixtures_set_raises_an_error_when_max_allowed_packet_is_smaller_than_fixtures_set_size
+    skip unless current_adapter?(:Mysql2Adapter)
+
+    begin
+      old_max_allowed_packet = ActiveRecord::Base.connection.execute("SELECT @@SESSION.max_allowed_packet").first[0]
+      ActiveRecord::Base.connection.execute("SET GLOBAL max_allowed_packet=1024")
+      fixtures = ["taggings", "posts", "companies", "comments"]
+
+      error = assert_raises(ActiveRecord::ActiveRecordError) { create_fixtures(*fixtures) }
+      assert_match(/Fixtures set is too large/, error.message)
+    ensure
+      ActiveRecord::Base.connection.execute("SET GLOBAL max_allowed_packet=#{old_max_allowed_packet}")
```

Comment:
Does this take effect immediately, or only upon reconnection? I think I saw a suggestion it could be the latter somewhere. That's fine for the earlier one, because we already reconnect to do the insertion.. but might mean we need another reconnect here?

Alternatively, it may be simpler to create new fixture(s) bigger than the default limit (using erb; no need for big actual files)?